pkgs/development/python-modules/playwright/default.nix

```diff
@@ -0,0 +1,247 @@
+    buildInputs = [ nodejs ];
+
+    postPatch = ''
+      # Use Nix's NodeJS instead of the bundled one.
+      substituteInPlace playwright.sh --replace '"$SCRIPT_PATH/node"' '"${nodejs}/bin/node"'
+      rm node
+
+      # Hard-code the script path to be our $out directory. The alternative would be to add
+      # coreutils to the closure and use realpath like the upstream script does.
+      substituteInPlace playwright.sh \
+        --replace 'SCRIPT_PATH="$(cd "$(dirname "$0")" ; pwd -P)"' "SCRIPT_PATH=$out"
+
+      patchShebangs playwright.sh package/bin/*.sh
+    '';
+
+    dontBuild = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin $out/share/playwright-driver
+      mv playwright.sh package $out/share/playwright-driver/
+      ln -s $out/share/playwright-driver/playwright.sh $out/bin/playwright
+
+      runHook postInstall
+    '';
+
+    passthru = {
+      inherit filename;
+    };
+  };
+
+  browsers-mac = stdenv.mkDerivation {
+    pname = "playwright-browsers";
+    version = driverVersion;
+
+    src = runCommand "playwright-browsers-base" {
+      outputHashMode = "recursive";
+      outputHashAlgo = "sha256";
+      outputHash = {
+        x86_64-darwin = "1nd6ll6dj0aqic3hh65y0br051ckbcap8spqbvzm19g8dn7vgv7m";
+      }.${system} or throwSystem;
+    } ''
+      export PLAYWRIGHT_BROWSERS_PATH=$out
+      ${driver}/playwright.sh install
+      rm -r $out/.links
+    '';
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir $out
+      cp -r * $out/
+    '';
+
+    doCheck = false;
+  };
+
+  browsers-linux = let
+    fontconfig = makeFontsConf {
+      fontDirectories = [];
+    };
+  in runCommand "playwright-browsers" {
+    nativeBuildInputs = [
+      makeWrapper
+      pkgs.jq # We want jq and not pythonPackages.jq
+    ];
+  } ''
+    BROWSERS_JSON=${driver}/share/playwright-driver/package/browsers.json
+    CHROMIUM_REVISION=$(jq -r '.browsers[] | select(.name == "chromium").revision' $BROWSERS_JSON)
+    FFMPEG_REVISION=$(jq -r '.browsers[] | select(.name == "chromium").revision' $BROWSERS_JSON)
+
+    mkdir -p \
+      $out/chromium-$CHROMIUM_REVISION/chrome-linux \
+      $out/ffmpeg-$FFMPEG_REVISION
+
+    # See here for the Chrome options:
+    # https://github.com/NixOS/nixpkgs/issues/136207#issuecomment-908637738
+    makeWrapper ${chromium}/bin/chromium $out/chromium-$CHROMIUM_REVISION/chrome-linux/chrome \
+      --set SSL_CERT_FILE /etc/ssl/certs/ca-bundle.crt \
+      --set FONTCONFIG_FILE ${fontconfig}
+
+    ln -s ${ffmpeg}/bin/ffmpeg $out/ffmpeg-$FFMPEG_REVISION/ffmpeg-linux
+  '';
+in
+buildPythonPackage rec {
+  pname = "playwright";
+  version = "1.25.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    sha256 = "sha256-K9hVTGGqTx3tTjGrwcLQisp3gq/lLJ/Y0GLDZcikSzw=";
+    leaveDotGit = true;
```

Comment:
Do we really must keep .git?

Reply:
Yeah, `setuptools-scm` needs it

Reply:
It seems common enough to use `SETUPTOOLS_SCM_PRETEND_VERSION` 
https://github.com/NixOS/nixpkgs/search?q=SETUPTOOLS_SCM_PRETEND_VERSION

Reply:
@ento yes, please use that suggestion.